### DISCUSSION TITLE
`onUpdate` doesnt get called by collection add/remove

### DISCUSSION THREAD

**Discussion by corrm** [mikro-orm/mikro-orm#6023] [Q&A]
```ts
@OneToMany(() => ReferralAction, action => action.referral, { onUpdate: () => { console.log("onUpdate"); } })
public readonly actions: Collection<ReferralAction>;

// Later
actions.add(new ReferralAction());
await em.flush();
```

`onUpdate` never hit. It should hit? if that not the way to do it what the correct one?

**Comment by B4nan** [comment]
No, it shouldn't, because nothing changed in the db table, there was no update query touching it. 1:m is an inverse side, what changes is the target entity (and that should get the event fired).

https://mikro-orm.io/docs/events#collections-and-on-update

**Comment by corrm** [comment]
Got the idea :+1: 

is there any way to make `Collection` readonly for public usage? i mean keep the add/remove logic for that entity class but prevent others from add/remove to the collection?

also is syntax valid?
```ts
@Entity()
class foo {
    @Property({ name: "bar" })
    private _bar: boolean = false;

    get bar(): boolean {
        return this._bar;
    }
    set bar(value: boolean) {
        this._bar = value;
    }
}
```

another question, can i do something like that on `populate` instead of use `referrals.load()`?
```ts
const user: User | null = await em.findOne(User, { id: session.user.id }, {
        populate: {
            referrals: { limit: 10, offset: 0 },
        },
    });
```

**Reply by B4nan** [reply]
> but prevent others 

Others as who? Other developers of the project you work on? I don't understand what you are up to. 

> also is syntax valid?

Well, yes and no, you can do this, but your property is still called `_bar` (the `name` option overrides the db column name, its alias for `fieldName`), and since TS won't allow us to infer private properties, you won't be able to use such property in partial loading (or populate hints if it would be a relation property), those parts are strictly type checked which requires properties to be defined as public. I would discourage this, you only make things more complex and break the type-safety on a lot of places.

> another question, can i do something like that on populate instead of use referrals.load()?

Nope, that's not supported, but you could use `Collection.matching()`. You can't paginate from populate hint, since that also works for `em.find` and I don't see any way to actually implement that for more than one entity.

**Comment by corrm** [comment]
> Others as who? Other developers of the project you work on? I don't understand what you are up to.

Other classes or functions outside the entity class

is there a repository pattern support? example?
And thanks for your time.
